_posts/2024-05-28-edb-and-kasten.markdown

```diff
@@ -0,0 +1,295 @@
+- **Migration and replication**  : we are more than ever in a hybrid momentum.
+
+Especially the last one **Migration and replication** :  no serious player let himself being trapped in a single cloud provider. How do you 
+negociate your bill if you cannot easily move ? How do you adapt to any change in the legal compliance of your country ?
+How do you move and replicate accross the cloud and the datacenters ? 
+ 
+EDB and Kasten together can overcome all these challenges let's see how. 
+ 
+# How EDB and Kasten work together ? 
+
+## EDB on Kubernetes 
+
+[EDB](https://www.enterprisedb.com) offer a full management of postgres clusters on Kubernetes. The only thing you have to do is create a custom resource like this one :
+
+```
+apiVersion: postgresql.k8s.enterprisedb.io/v1
+kind: Cluster
+metadata:
+  name: cluster-example
+  annotations:
+    "k8s.enterprisedb.io/addons": '["kasten"]'    
+spec:
+  instances: 3
+  storage:
+    size: 1Gi
+ ```
+ 
+"Et Voila", you immediately get a postgres cluster with one primary and 2 replicas. If the primary fail then the failover to one of the replica is immediate. Not only the operator will handle the failover it will also take care of repairing by creating another replicas so that you do not stay 
+in a degraded situation. All that automatically.
+ 
+EDB is the company behind the very well known Open Source Kubernetes project [CloudNativePG](https://cloudnative-pg.io), and you can get support and professional services from EDB through their [commercial offering](https://www.enterprisedb.com/services-support/professional-services). You can see 
+an EDB database as a managed database but without being tied in a cloud provider or a data center.
+
+
+## Solution 
+
+In it's regular run EDB manage the postgres cluster and make sure one instance is primary (read-write) and 2 replicas (read-only) are available in 
+case of failure to do the failover.
+
+![regular run](/images/posts/2024-05-28-edb-and-kasten/regular-run.png)
+
+When Kasten launch the backup it call the command to "Fence" one of the replica. It means that the primary instance is not affected and all the transactions can continue on the primary. Customer won't notice any downtime or performance drop because of the backup ! 
+
+![Before snapshot](/images/posts/2024-05-28-edb-and-kasten/before-snapshot.png)
+
+When an intance is "Fenced" then it is garanteed to be consistent, which means that all transaction has been properly commited before the fence command
+return so now Kasten is 100% sure to take a fully consistent snapshot of the database.
+
+When the snapshot of the "Fenced" instance is over then Kasten will "Unfence" it and the instance will catch up with the last transactions that he eventually missed during the fencing.
+
+![After snapshot](/images/posts/2024-05-28-edb-and-kasten/after-snapshot.png)
+
+Now when it comes to restore, Kasten will only restore the PVC that was fenced and the operator will promote this instance primary 
```

Comment:
During restore, Kasten restores the PVC that was fenced, and when complete the operator promotes it as a primary instance. Two read replicas are then created from the primary.